### PR TITLE
haskellPackages.glpk-hs: fix for ghc-8.6

### PR DESCRIPTION
**Pull request by pschuprikov**
###### Motivation for this change
From what I was able to understand, the hackage version of __glpk-hs__ does not work with the new __gasp__. Moreover, due to deprecated calls __glpk-hs__ doesn't compile with __ghc-8.6__, which is fixed by the patch.

The last [upstream](https://github.com/jyp/glpk-hs) change was 8 months ago.

###### Things done

Compiles fine with __ghc-8.4.4__, __ghc-8.6.5__, __ghc-8.8.1__.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by turion** [comment]
There has been a release to version 0.7 a month ago. Is this still needed?

**Comment by pschuprikov** [comment]
There are still deprecation errors. A pending upstream PR https://github.com/jyp/glpk-hs/pull/11 from which I've taken the patch hasn't been merged yet. I'll remove `overrideSrc` part then.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1327,4 +1327,9 @@ self: super: {
   haskell-lsp_0_18_0_0 = super.haskell-lsp_0_18_0_0.override { haskell-lsp-types = self.haskell-lsp-types_0_18_0_0; };
   lsp-test_0_8_2_0 = (dontCheck super.lsp-test_0_8_2_0).override { haskell-lsp = self.haskell-lsp_0_18_0_0; };
 
+  # https://github.com/jyp/glpk-hs/pull/11
+  # patch is the above PR with conflicts resolved
+  glpk-hs = assert super.glpk-hs.version == "0.7"; 
+    appendPatch super.glpk-hs ./patches/glpk-hs-ghc86.patch;
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix]: If possible, in nixpkgs we generally prefer to pull patches from upstream directly with `fetchpatch`.  There should be some examples of doing this in this file.

Can you rewrite this to use fetchpatch?

**Comment by cdepillabout** [comment]
@pschuprikov Can you rebase this on the `haskell-updates` branch?

File: pkgs/development/haskell-modules/configuration-common.nix (hunk 1 above)
Comment (pschuprikov) [reply pkgs/development/haskell-modules/configuration-common.nix]: That was exactly my previous approach, but the PR in its current form has merge conflicts with 0.7 and the patching fails. If there is a cleaner solution other than creating a patch manually, I would certainly use it.

Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix]: @pschuprikov Could you send the patch in this PR as a PR to upstream?

**Comment by peti** [comment]
Please submit a cleaned-up version of the change if you're still interested in fixing this issue.

**Pull request closed without merging**